src/stores/ImageFittingStore.ts

```diff
@@ -0,0 +1,157 @@
+        return [{value: ACTIVE_FILE_ID, label: "Active"}, ...(AppStore.Instance.frameNames ?? [])];
+    }
+
+    @computed get effectiveFrame(): FrameStore {
+        const appStore = AppStore.Instance;
+        if (appStore.activeFrame && appStore.frames?.length > 0) {
+            return this.selectedFileId === ACTIVE_FILE_ID ? appStore.activeFrame : appStore.getFrame(this.selectedFileId) ?? appStore.activeFrame;
+        }
+        return null;
+    }
+
+    @computed get fitDisabled() {
+        const validFileId = this.effectiveFrame?.frameInfo && this.effectiveFrame?.frameInfo?.fileId >= 0;
+        const validParams = this.components.every(c => c.validParams === true);
+        return !(validFileId && validParams) || this.isFitting;
+    }
+
+    constructor() {
+        makeObservable(this);
+        this.selectedFileId = ACTIVE_FILE_ID;
+        this.clearComponents();
+        this.selectedComponentIndex = 0;
+    }
+
+    fitImage = () => {
+        if (this.fitDisabled) {
+            return;
+        }
+        this.setIsFitting(true);
+        const initialValues = [];
+        for (const c of this.components) {
+            initialValues.push({
+                centerX: c.center.x,
+                centerY: c.center.y,
+                amp: c.amplitude,
+                fwhmX: c.fwhm.x,
+                fwhmY: c.fwhm.y,
+                pa: c.pa,
+                fixedCenterX: false,
+                fixedCenterY: false,
+                fixedAmp: false,
+                fixedFwhmX: false,
+                fixedFwhmY: false,
+                fixedPa: false
+            });
+        }
+
+        const message: CARTA.IFittingRequest = {
+            fileId: this.effectiveFrame.frameInfo.fileId,
+            initialValues: initialValues
+        };
+        AppStore.Instance.requestFitting(message);
+    };
+}
+
+export class ImageFittingIndividualStore {
+    @observable center: Point2D;
+    @observable amplitude: number;
+    @observable fwhm: Point2D;
+    @observable pa: number;
+
+    @action setCenterX = (val: number) => {
+        this.center.x = val;
+    };
+
+    @action setCenterY = (val: number) => {
+        this.center.y = val;
+    };
+
+    @action setAmplitude = (val: number) => {
+        this.amplitude = val;
+    };
+
+    @action setFwhmX = (val: number) => {
+        this.fwhm.x = val;
+    };
+
+    @action setFwhmY = (val: number) => {
+        this.fwhm.y = val;
+    };
+
+    @action setPa = (val: number) => {
+        this.pa = val;
+    };
+
+    constructor() {
+        makeObservable(this);
+        this.center = {x: null, y: null};
+        this.amplitude = null;
+        this.fwhm = {x: null, y: null};
+        this.pa = null;
+    }
+
+    @computed get validParams() {
+        return Number.isFinite(this.center.x) && Number.isFinite(this.center.y) && Number.isFinite(this.amplitude) && Number.isFinite(this.fwhm.x) && Number.isFinite(this.fwhm.y) && Number.isFinite(this.pa);
```

Comment:
can't you just use `isFinite` without `Number.`? Perhaps just in case include `this.center && this.fwhm &&` to ensure the points exist

Reply:
I used `Number.isFinite` because `isFinite(null)` returns true. I think we can change the initial value from `null` to `NaN` as you suggested, and change here to `isFinite`

Reply:
changed to `isFinite` and added optional chaining to check if the points exist